src/main/java/com/algorand/algosdk/account/Account.java

```diff
@@ -514,23 +555,41 @@ public LogicsigSignature appendToLogicsig(LogicsigSignature lsig) throws Illegal
 
     }
 
+    public static SignedTransaction signLogicTransactionWithAddress(LogicsigSignature lsig, Address lsigAddr, Transaction tx)
+            throws NoSuchAlgorithmException, IOException {
```

Comment:
Do you need to declare `NoSuchAlgorithmException` here? It was probably added for creating `SignedTransaction`, but you convert it to an IOException.

Reply:
Yes, it comes from the `verify` in logicSigSignature, any modification needed on this?

Reply:
It seems like we need to catch `NoSuchAlgorithmException` and wrap it in another exception to maintain backwards compatibility?

Reply:
Yea, I think for both signLogicTransaction functions, we need to catch `NoSuchAlgorithmException`, but I am not sure about if anything need to be done on `verify` function.